pkgs/tools/system/nvtop/build-nvtop.nix

```
@@ -0,0 +1,97 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, gtest
+, cudatoolkit
+, libdrm
+, ncurses
+, testers
+, udev
+, addOpenGLRunpath
+, amd ? false
+, intel ? false
+, msm ? false
+, nvidia ? false
+, apple ? false
+, panfrost ? false
+, panthor ? false
+, ascend ? false
+}:
+
+let
+  nvidia-postFixup = "addOpenGLRunpath $out/bin/nvtop";
+  libPath = lib.makeLibraryPath [ libdrm ncurses udev ];
+  drm-postFixup = ''
+    patchelf \
+      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${libPath}" \
+      $out/bin/nvtop
+  '';
+  needDrm = (amd || msm || panfrost || panthor);
+  onOff = v: if v then "ON" else "OFF";
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "nvtop";
+  version = "3.1.0";
+
+  src = fetchFromGitHub {
+    owner = "Syllo";
+    repo = "nvtop";
+    rev = finalAttrs.version;
+    hash = "sha256-MkkBY2PR6FZnmRMqv9MWqwPWRgixfkUQW5TWJtHEzwA=";
+  };
+
+  cmakeFlags = [
+    "-DBUILD_TESTING=ON"
+    "-DUSE_LIBUDEV_OVER_LIBSYSTEMD=ON"
+    "-DAMDGPU_SUPPORT=${onOff amd}"
+    "-DNVIDIA_SUPPORT=${onOff nvidia}"
+    "-DINTEL_SUPPORT=${onOff intel}"
+    "-DAPPLE_SUPPORT=${onOff apple}"
+    "-DMSM_SUPPORT=${onOff msm}"
+    "-DPANFROST_SUPPORT=${onOff panfrost}"
+    "-DPANTHOR_SUPPORT=${onOff panthor}"
+    "-DASCEND_SUPPORT=${onOff ascend}"
+  ] ++ lib.optionals nvidia [
+    "-DNVML_INCLUDE_DIRS=${cudatoolkit}/include"
+    "-DNVML_LIBRARIES=${cudatoolkit}/targets/x86_64-linux/lib/stubs/libnvidia-ml.so"
+  ];
+
+  nativeBuildInputs = [ cmake gtest ] ++ lib.optional nvidia addOpenGLRunpath;
+
+  buildInputs = with lib; [ ncurses udev ]
+    ++ optional nvidia cudatoolkit
+    ++ optional needDrm libdrm
+  ;
```

I don't like this formatting, and I'm don't remember the rules now that https://github.com/NixOS/rfcs/pull/166 is merged, but I'm pretty sure that this should be formatted to:

```suggestion
  nativeBuildInputs = [
    cmake
    gtest
  ] ++ lib.optionals nvidia [
    addOpenGLRunpath
  ];

  buildInputs = [
    ncurses
    udev
  ] ++ lib.optional nvidia [
    cudatoolkit
  ] ++ optional needDrm [
    libdrm
  ];
```

---

No need to manually point out how to format things. Once the RFC is implemented, CI will do that, and until then it doesn't really matter how it's formatted imo, as long as it's readable :)

---

I'm using nixpkgs-fmt for formatting, IMO the file is readable enough. Let's wait for official formatting validation and then fix any discrepancies that will be found.